services/scheduler/service.go

```diff
@@ -22,6 +22,9 @@ import (
 // JobFunc is the type for jobs.
 type JobFunc func(context.Context, interface{})
 
+// JobFuncNoData is the type for jobs.
+type JobFuncNoData func(context.Context)
```

Comment:
Do we still need `JobFuncNoData`, or can we make this `JobFunc` and remove the existing definition?

Reply:
`JobFunc` is needed for the periodic schedule method. I have a PR ready to make similar changes to the periodic schedule method and that will also remove `JobFuncNoData`.